Reduce Azure OpenAI RPM per replica and limit max replicas to prevent rate limiting

## Problem

The bias-scoring-service is hitting Azure OpenAI rate limits (429 errors) during scans. After 5 retries, requests fail with 503 Service Unavailable, causing pages to not get LLM scored.

From the logs:
```
[RATE LIMIT] 429 received, retrying in 1.3s (attempt 1/5)
[RATE LIMIT] 429 received, retrying in 2.7s (attempt 2/5)
[RATE LIMIT] 429 received, retrying in 4.3s (attempt 3/5)
[RATE LIMIT] 429 received, retrying in 8.4s (attempt 4/5)
[ERROR] Rate limit exhausted after 5 retries
INFO: POST /score_page HTTP/1.1 503 Service Unavailable
```

## Proposed Solution

With a 20 RPM Azure OpenAI quota, configure 2 replicas at 10 RPM each:

1. **Add `AZURE_OPENAI_RPM` environment variable** to limit requests per replica:
```yaml
- name: AZURE_OPENAI_RPM
  value: "10"  # Split 20 RPM quota across 2 replicas
```

2. **Set `maxReplicaCount: 2`** in the KEDA ScaledObject to prevent scaling beyond quota

## Files to Update

- `infra/k8s/base/bias-scoring-service-keda.yaml`

## Context

Discovered while investigating scan 95 where pages were not getting LLM scored due to rate limit exhaustion.

--- a/infra/k8s/base/bias-scoring-service-keda.yaml
+++ b/infra/k8s/base/bias-scoring-service-keda.yaml
@@ -34,9 +34,11 @@ spec:
       containers:
         - name: bias-scoring-service
           image: seanmckdemo.azurecr.io/bias-scoring-service
-          env:            
+          env:
             - name: AZURE_OPENAI_DEPLOYMENT
               value: "gpt-4.1"
+            - name: AZURE_OPENAI_RPM
+              value: "10"
           envFrom:
             - secretRef:
                 name: sc-linuxdocsazureopenai-secret
@@ -73,7 +75,7 @@ spec:
   scaleTargetRef:
     name: bias-scoring-service
   minReplicaCount: 2
-  maxReplicaCount: 10
+  maxReplicaCount: 2
   cooldownPeriod: 60
   triggers:
   - type: cpu

